### ISSUE TITLE
AOTAutograd export path does not support training graphs with parameters that do not receive gradients.

### ISSUE THREAD

**Issue by bdhirsh** [pytorch/pytorch#101192]
See the comment [here](https://github.com/pytorch/pytorch/pull/100587/files#diff-df954bbf954d2dcb81f687876053267ffa4ddb36ed86b7d2bd76319ff2b94416R3786): today, the AOTAutograd export path expects that every parameter in the module you export will receive some sort of gradient. It's possible to create a case where this doesn't happen. We can support it, but I'd like to gauge how much request there is for this first, since it's more of an edge case.

Example:

```
def f(p1, p2):
    # p1 will never receive any gradients. p2 will though
    return p1.detach() * p2
```

**Comment by pianpwk** [comment]
we had this issue come up for the joint-graph API, for tied weights: https://github.com/pytorch/pytorch/issues/147380

**Comment by wmhst7** [comment]
This behavior is becoming increasingly relevant with the prevalence of Mixture-of-Experts (MoE) architectures. In models like Qwen3 MoE, for example, only a small subset of experts (e.g., 8 out of 128) are activated and receive gradients during a given forward pass; the weights of inactive experts intentionally do not.

Therefore, I anticipate a need for AOTAutograd to support modules where not all parameters are guaranteed to receive gradients.  I'm also open to discussing any alternative workarounds. Thanks!

cc @bdhirsh

**Comment by bdhirsh** [comment]
@wmhst7 do you have a link to some MoE example code with that behavior?

**Comment by wmhst7** [comment]
@bdhirsh [Qwen3MoeSparseMoeBlock](https://github.com/huggingface/transformers/blob/51d732709e5ae424e8fb6c4e58b72057a3e413c2/src/transformers/models/qwen3_moe/modeling_qwen3_moe.py#L215)
